files/scripts/startup_tsa_tsb.py

```diff
@@ -0,0 +1,156 @@
+#!/usr/bin/env python3
+
+# Name: startup_tsa_tsb.py, version: 1.0
+#
+# Description: Module contains the definitions to the VOQ Startup TSA-TSB service
+from sonic_py_common import multi_asic, device_info
+from sonic_py_common.logger import Logger
+import subprocess
+import sys, getopt
+from threading import Timer
+import os
+
+# Global Logger class instance
+logger = Logger("startup_tsa_tsb")
+logger.set_min_log_priority_info()
+
+def get_tsb_timer_interval():
+    platform = device_info.get_platform()
+    conf_file = '/usr/share/sonic/device/{}/startup-tsa-tsb.conf'.format(platform)
+    file = open(conf_file, 'r')
+    Lines = file.readlines()
+    for line in Lines:
+        field = line.split('=')[0].strip()
+        if field == "STARTUP_TSB_TIMER":
+            return line.split('=')[1].strip()
+    return 0
+
+def get_sonic_config(ns, config_name):
+    if ns == "":
+        return subprocess.check_output(['sonic-cfggen', '-d', '-v', config_name.replace('"', "'"), ]).strip()
+    else:
+        return subprocess.check_output(['sonic-cfggen', '-d', '-v', config_name.replace('"', "'"), '-n', ns.replace('"', "'")]).strip()
+
+def get_sub_role(asic_ns):
+    sub_role_config = "DEVICE_METADATA['localhost']['sub_role']"
+    sub_role = (get_sonic_config(asic_ns, sub_role_config)).decode()
+    return sub_role
+
+def get_tsa_config(asic_ns):
+    tsa_config = 'BGP_DEVICE_GLOBAL.STATE.tsa_enabled'
+    tsa_ena = (get_sonic_config(asic_ns, tsa_config)).decode()
+    if asic_ns == "":
+        logger.log_info('CONFIG_DB.{} : {}'.format(tsa_config, tsa_ena))
+    else:
+        logger.log_info('{} - CONFIG_DB.{} : {}'.format(asic_ns, tsa_config, tsa_ena))
+    return tsa_ena
+
+def get_tsa_status(num_asics):
+    if num_asics > 1:
+        counter = 0
+        for asic_id in range(int(num_asics)):
+            asic_ns = 'asic{}'.format(asic_id)
+            sub_role = get_sub_role(asic_ns)
+            if sub_role == 'FrontEnd':
+                tsa_enabled = get_tsa_config(asic_ns)
+                if tsa_enabled == 'false':
+                    counter += 1
+        if counter == int(num_asics):
+            return True;
+    else:
+        tsa_enabled = get_tsa_config("")
+        if tsa_enabled == 'false':
```

Comment:
why is tsa_enabled not returned as it is? The function return value is confusing

Reply:
Tejaswini, The startup-tsa-tsb service is global and the checks in get_tsa_status makes sure that the service is started only when the tsa_enabled flag is set to false for all namespaces in multi-asic and in the global config for pizza box.